test/libsolidity/smtCheckerTests/control_flow/revert.sol

```diff
@@ -0,0 +1,17 @@
+pragma experimental SMTChecker;
+
+contract C {
+	function f() pure public {
+		require(false);
+		// This is not reachable.
+		assert(false);
+	}
+
+	function g() pure public {
+		revert();
+		// This is not reachable.
+		assert(false);
+	}
+}
+// ----
+// Warning 5740: (211-224): Unreachable code.
```

Comment:
This comes from `analysis/ControlFlowAnalyzer` which apparently is oblivious to revert.

cc @chriseth @ekpyron

Reply:
This is exactly right, isn't it? The control flow analyzer doesn't perform data flow analysis (at all), so it won't recognize ``false`` being always false  in ``require(false)``, hence it won't complain about the ``f()``, but in ``g()`` it **does** recognize ``revert()`` and correctly complains that ``assert(false)`` is unreachable - so not sure what's the problem.

Reply:
It actually recognised `require(false)` and does not recognise `revert`.

Reply:
Nope!

Reply:
The warning happens, if it does recognize it (because only *then* the following code *is* detected to be unreachable) - and I just ran it (albeit on a different branch) and it does warn for the assert after the ``revert()``, i.e. it recognizes revert and doesn't recognize ``require(false)``.

Reply:
contract C {
	function f() pure public {
		require(false);
		// This is not reachable.
		assert(false);
	}

	function g() pure public {
		revert();
		// This is not reachable.
		assert(false);
	}
}

Reply:
We could argue about refining control flow analysis for cases like this separately from this PR, but actually this is behaving as intended so far.

Reply:
> It actually recognised require(false) and does not recognise revert.

Checked again and I remembered the highlighting isoltest does incorrectly, it only evaluates the assertions after revert as unreachable.